### PR TITLE
Add support for Focalboard, newer Apache and SSL

### PR DESCRIPTION
**Pull request by anx-ag**
Based on some user complaints in the forum and the community channels, I had a look at the Apache documentation here and it was missing support for the focalboard websockets as well as SSL and newer mod_proxy_http-only setups on Apache 2.4.47+.

I did also rewrite some other parts to be more specific and give hints to the commands necessary.

I have no idea how to move the bulletpoint "3" at the end out of the codeblock, if someone could fix that for me (and let me know what I did wrong), this would be awesome :) Please read through it, an additional pair of eyes would be good - thanks!

The new configuration has been tested and verified working on one of my demo systems, I did not test the old config but assume it was working and still is.

### CODE REVIEW

**Comment by cwarnermm** [comment]
Thanks, @anx-ag! I'm happy to help with the RST syntax needed for the codeblock and bullet points. I've reached out to our Focalboard experts in Engineering to request a technical review of this docs PR.

File: source/configure/config-proxy-apache2.rst [hunk 1]
```diff
@@ -42,16 +99,15 @@ On a Debian-based operating system such as Ubuntu, Apache2 proxy configuration i
...
 
-5. (Debian/Ubuntu only) Because you'll likely have not set up the subdomain before now on Apache2, run ``a2ensite mysubdomain.mydomain.com`` to enable the site (do not run ``a2ensite mysubdomain.mydomain.com.conf``).
-
-6. Restart Apache2.
-
-    - On Ubuntu 14.04 and RHEL 6: ``sudo service apache2 restart``
-    - On Ubuntu 16.04+ and RHEL 7+: ``sudo systemctl restart apache2``
-
-You should be all set! Ensure that your Mattermost config file is pointing to the correct URL (which may include a port), and then ensure that your socket connection is not dropping once deployed. To prevent external access to Mattermost on port 8065, in the config file, set ``ListenAddress`` to ``localhost:8065`` instead of ``:8065``.
+ 3. If you have enabled TLS in the Mattermost settings, you must also enable the SSL module in Apache (usually with ``a2enmod ssl``) and change the ``http://`` part of the proxy destination to ``https://`` and the websocket protocol from ``ws://`` to ``wss://`` additionally to enabling the ``SSLProxyEngine``.
```
Comment (cwarnermm) [inline source/configure/config-proxy-apache2.rst:113]: @anx-ag - I suspect that the single blank space immediately before step 3 is what's causing the step text to display as part of the code block:

Screenshot 2023-03-03 at 10 59 03 AM

File: source/configure/config-proxy-apache2.rst [hunk 2]
```diff
@@ -2,38 +2,95 @@
...
 
 1. SSH into your server.
-2. Make sure the Apache modules ``mod_rewrite`` , ``mod_proxy``, ``mod_proxy_http``, and ``mod_proxy_wstunnel`` are installed and enabled. If not, follow the instructions from your Linux distribution to do so.
+2. Make sure the Apache modules ``mod_rewrite``, ``mod_proxy`` and ``mod_proxy_http`` are installed and enabled. If not, follow the instructions from your Linux distribution to do so (usually this is being done with ``a2enmod rewrite``, ``a2enmod proxy`` and ``a2enmod proxy_http``).
 3. Create the above mentioned configuration file. It is often helpful to start with a copy of ``000-default.conf`` or ``default-ssl.conf`` (on Ubuntu).
 4. Edit your configuration using the guide below:
 
-    1. If you're not setting up a subdomain, your ``ServerName`` will simply be set to ``mydomain.com``.
-    2. ``ServerAlias`` can been added too if you want to capture ``www.mydomain.com``.
+    1. If you're not setting up a subdomain, your ``ServerName`` will simply be set to ``mymattermost.tld``.
```
Comment (cwarnermm) [inline source/configure/config-proxy-apache2.rst:19]: @anx-ag - These sub-steps are currently displaying as 1., 2., 3., and should display as a., b., c. instead. You should be able to replace the numbers with alpha characters inline directly to address this.

File: source/configure/config-proxy-apache2.rst (hunk 1 above)
Comment (anx-ag) [reply source/configure/config-proxy-apache2.rst:113]: Oh... Sorry for that :-( Should I resubmit or can you fix that quickly?

File: source/configure/config-proxy-apache2.rst [hunk 3]
```diff
@@ -2,38 +2,95 @@
...
+          
+          SSLEngine On
+          SSLCertificateFile /path/to/your/certificate.pem
+          SSLCertificateKeyFile /path/to/your/certificate.key
+          
+          # Uncomment the following line if your Mattermost server is requiring SSL/TLS connections
+          # SSLProxyEngine On
+        </VirtualHost>
+
+6. Restart Apache2.
```
Comment (cwarnermm) [inline source/configure/config-proxy-apache2.rst:58]: Step 6 currently doesn't align to either the initial set of steps nor the substeps, so I'm unclear about which set of steps it applies to. 

If it's truly step 6 and applies to the substeps, please adjust the indentation and update the step from 6 to f.

And if it's actually step 5, it's fine as-is from an indentation perspective, but needs to be renumbered.

File: source/configure/config-proxy-apache2.rst (hunk 1 above)
Comment (cwarnermm) [reply source/configure/config-proxy-apache2.rst:113]: If you're open to addressing the issues I've identified in this PR as a batch, feel free to do so. Let me know if you'd like my assistance with these formatting issues, and I can submit changes.

Comment (anx-ag) [reply source/configure/config-proxy-apache2.rst:113]: Thanks, removing the space fixed it.

File: source/configure/config-proxy-apache2.rst (hunk 2 above)
Comment (anx-ag) [reply source/configure/config-proxy-apache2.rst:19]: I did replace them with alphabet characters, the preview still shows roman numbers - let's see if it works in production then :)

File: source/configure/config-proxy-apache2.rst (hunk 3 above)
Comment (anx-ag) [reply source/configure/config-proxy-apache2.rst:58]: Fixed, there is no step 6 anymore - it's 5.

**Comment by cwarnermm** [comment]
Thanks, @anx-ag, for creating this docs PR. Unfortunately, we won't be able to merge it. This docs page appears live, but was long ago archived and removed from docs builds because Apache 2.0 isn't officially supported.

Mattermost product documentation only features officially supported features and functionality. Anything unofficial or community-driven can be accessed via the [Forum](https://forum.mattermost.com/c/docs/37).

This archived page appears to be live because it exists on the production server in error, and is accessible by URL. I'm working with DevOps to address this issue with the production server.

I really appreciate your efforts to ensure our product documentation is accurate, complete, and maintained. Do any of the changes you've proposed here apply to the [Forum post](https://forum.mattermost.com/t/configuring-apache2-as-a-proxy-for-mattermost-server/11938/2)? If so, would you be open to updating that post?

**Pull request closed without merging**